--- libav27416.org/configure	2011-01-27 05:23:42.000000000 +0900
+++ libav27416/configure	2011-01-27 05:45:10.000000000 +0900
@@ -2583,8 +2583,8 @@
     # AltiVec flags: The FSF version of GCC differs from the Apple version
     if enabled altivec; then
         nogas=warn
-        check_cflags -maltivec -mabi=altivec &&
-        { check_header altivec.h && inc_altivec_h="#include <altivec.h>" ; } ||
+#       check_cflags -maltivec -mabi=altivec &&
+#       { check_header altivec.h && inc_altivec_h="#include <altivec.h>" ; } ||
         check_cflags -faltivec
 
         # check if our compiler supports Motorola AltiVec C API
