--- libav27703.org/configure	2011-02-21 21:05:54.000000000 +0900
+++ libav27703/configure	2011-02-21 21:07:16.000000000 +0900
@@ -2593,8 +2593,8 @@
     # AltiVec flags: The FSF version of GCC differs from the Apple version
     if enabled altivec; then
         nogas=warn
-        check_cflags -maltivec -mabi=altivec &&
-        { check_header altivec.h && inc_altivec_h="#include <altivec.h>" ; } ||
+#       check_cflags -maltivec -mabi=altivec &&
+#       { check_header altivec.h && inc_altivec_h="#include <altivec.h>" ; } ||
         check_cflags -faltivec
 
         # check if our compiler supports Motorola AltiVec C API
