--- libav28543.org/configure	2011-05-15 12:32:53.000000000 +0900
+++ libav28543/configure	2011-05-15 12:46:27.000000000 +0900
@@ -2635,8 +2635,8 @@
     # AltiVec flags: The FSF version of GCC differs from the Apple version
     if enabled altivec; then
         nogas=warn
-        check_cflags -maltivec -mabi=altivec &&
-        { check_header altivec.h && inc_altivec_h="#include <altivec.h>" ; } ||
+#       check_cflags -maltivec -mabi=altivec &&
+#       { check_header altivec.h && inc_altivec_h="#include <altivec.h>" ; } ||
         check_cflags -faltivec
 
         # check if our compiler supports Motorola AltiVec C API
