--- libav25146.org/configure	2010-09-20 07:31:11.000000000 +0900
+++ libav25146/configure	2010-09-20 07:44:43.000000000 +0900
@@ -2532,8 +2532,8 @@
     # AltiVec flags: The FSF version of GCC differs from the Apple version
     if enabled altivec; then
         nogas=warn
-        check_cflags -maltivec -mabi=altivec &&
-        { check_header altivec.h && inc_altivec_h="#include <altivec.h>" ; } ||
+#       check_cflags -maltivec -mabi=altivec &&
+#       { check_header altivec.h && inc_altivec_h="#include <altivec.h>" ; } ||
         check_cflags -faltivec
 
         # check if our compiler supports Motorola AltiVec C API
