--- x2641947.org/configure	2011-04-26 19:38:57.000000000 +0900
+++ x2641947/configure	2011-04-26 19:48:02.000000000 +0900
@@ -483,7 +483,7 @@
             define HAVE_ALTIVEC
             AS="${AS-${cross_prefix}gcc}"
             if [ $SYS = MACOSX ] ; then
-                CFLAGS="$CFLAGS -faltivec -fastf -mcpu=G4"
+                CFLAGS="$CFLAGS -faltivec -fastf -mcpu=G4 -mtune=7450"
             else
                 CFLAGS="$CFLAGS -maltivec -mabi=altivec"
                 define HAVE_ALTIVEC_H
@@ -1008,8 +1008,8 @@
 
 filters="crop select_every"
 gpl_filters=""
-[ $swscale = yes ] && filters="resize $filters"
-[ $gpl = yes ] && filters="$filters $gpl_filters"
+[ "$swscale" = "yes" ] && filters="resize $filters"
+[ "$gpl" = "yes" ] && filters="$filters $gpl_filters"
 
 cat > conftest.log <<EOF
 Platform:   $ARCH
