--- x2641680.org/configure	2010-07-20 09:09:55.000000000 +0900
+++ x2641680/configure	2010-07-20 09:18:50.000000000 +0900
@@ -367,7 +367,7 @@
       define HAVE_ALTIVEC
       AS="${AS-${cross_prefix}gcc}"
       if [ $SYS = MACOSX ] ; then
-        CFLAGS="$CFLAGS -faltivec -fastf -mcpu=G4"
+        CFLAGS="$CFLAGS -faltivec -fastf -mcpu=G4 -mtune=7450"
       else
         CFLAGS="$CFLAGS -maltivec -mabi=altivec"
         define HAVE_ALTIVEC_H
@@ -766,7 +766,7 @@
 EOF
 
 filters="crop select_every"
-[ $swscale = yes ] && filters="resize $filters"
+[ "$swscale" = "yes" ] && filters="resize $filters"
 
 cat > conftest.log <<EOF
 Platform:   $ARCH
