--- x2641788.org/configure	2010-11-20 09:52:19.000000000 +0900
+++ x2641788/configure	2010-11-20 09:54:08.000000000 +0900
@@ -384,7 +384,7 @@
       define HAVE_ALTIVEC
       AS="${AS-${cross_prefix}gcc}"
       if [ $SYS = MACOSX ] ; then
-        CFLAGS="$CFLAGS -faltivec -fastf -mcpu=G4"
+        CFLAGS="$CFLAGS -faltivec -fastf -mcpu=G4 -mtune=7450"
       else
         CFLAGS="$CFLAGS -maltivec -mabi=altivec"
         define HAVE_ALTIVEC_H
@@ -805,8 +805,8 @@
 
 filters="crop select_every"
 gpl_filters=""
-[ $swscale = yes ] && filters="resize $filters"
-[ $gpl = yes ] && filters="$filters $gpl_filters"
+[ "$swscale" = "yes" ] && filters="resize $filters"
+[ "$gpl" = "yes" ] && filters="$filters $gpl_filters"
 
 cat > conftest.log <<EOF
 Platform:   $ARCH
