--- x2641882.org/configure	2011-01-28 05:41:16.000000000 +0900
+++ x2641882/configure	2011-01-28 05:43:01.000000000 +0900
@@ -391,7 +391,7 @@
             define HAVE_ALTIVEC
             AS="${AS-${cross_prefix}gcc}"
             if [ $SYS = MACOSX ] ; then
-                CFLAGS="$CFLAGS -faltivec -fastf -mcpu=G4"
+                CFLAGS="$CFLAGS -faltivec -fastf -mcpu=G4 -mtune=7450"
             else
                 CFLAGS="$CFLAGS -maltivec -mabi=altivec"
                 define HAVE_ALTIVEC_H
@@ -842,8 +842,8 @@
 
 filters="crop select_every"
 gpl_filters=""
-[ $swscale = yes ] && filters="resize $filters"
-[ $gpl = yes ] && filters="$filters $gpl_filters"
+[ "$swscale" = "yes" ] && filters="resize $filters"
+[ "$gpl" = "yes" ] && filters="$filters $gpl_filters"
 
 cat > conftest.log <<EOF
 Platform:   $ARCH
