--- x2641666.org/configure	2010-07-09 22:20:04.000000000 +0900
+++ x2641666/configure	2010-07-10 15:44:23.000000000 +0900
@@ -343,7 +343,7 @@
       define HAVE_ALTIVEC
       AS="${AS-${cross_prefix}gcc}"
       if [ $SYS = MACOSX ] ; then
-        CFLAGS="$CFLAGS -faltivec -fastf -mcpu=G4"
+        CFLAGS="$CFLAGS -faltivec -fastf -mcpu=G4 -mtune=7450"
       else
         CFLAGS="$CFLAGS -maltivec -mabi=altivec"
         define HAVE_ALTIVEC_H
