--- x2642120.org/configure	2011-12-08 06:53:41.000000000 +0900
+++ x2642120/configure	2011-12-25 14:23:03.000000000 +0900
@@ -626,7 +626,7 @@
     fi
 fi
 
-if [ $shared = yes -a \( $ARCH = "X86_64" -o $ARCH = "PPC" -o $ARCH = "ALPHA" -o $ARCH = "ARM" -o $ARCH = "IA64" -o $ARCH = "PARISC" -o $ARCH = "MIPS" \) ] ; then
+if [ $shared = "yes" -a \( $ARCH = "X86_64" -o $ARCH = "PPC" -o $ARCH = "ALPHA" -o $ARCH = "ARM" -o $ARCH = "IA64" -o $ARCH = "PARISC" -o $ARCH = "MIPS" \) ] ; then
     pic="yes"
 fi
 
@@ -933,9 +933,9 @@
 
 ASFLAGS="$ASFLAGS -DBIT_DEPTH=$bit_depth"
 
-[ $gpl = yes ] && define HAVE_GPL && x264_gpl=1 || x264_gpl=0
+[ $gpl = "yes" ] && define HAVE_GPL && x264_gpl=1 || x264_gpl=0
 
-[ $interlaced = yes ] && define HAVE_INTERLACED && x264_interlaced=1 || x264_interlaced=0
+[ $interlaced = "yes" ] && define HAVE_INTERLACED && x264_interlaced=1 || x264_interlaced=0
 
 #define undefined vars as 0
 for var in $CONFIG_HAVE; do
@@ -953,7 +953,7 @@
     LIBX264=libx264.lib
     RANLIB=
     STRIP=
-    if [ $debug = yes ]; then
+    if [ $debug = "yes" ]; then
         LDFLAGS="-debug $LDFLAGS"
         CFLAGS="-D_DEBUG $CFLAGS"
     else
@@ -974,7 +974,7 @@
 else
     CFLAGS="$(intel_cflags $CFLAGS)"
     # icc does not define __SSE__ until SSE2 optimization and icl never defines it or _M_IX86_FP
-    [ \( $ARCH = X86_64 -o $ARCH = X86 \) -a $asm = yes ] && ! cpp_check "" "" "defined(__SSE__)" && define __SSE__
+    [ \( $ARCH = X86_64 -o $ARCH = X86 \) -a $asm = "yes" ] && ! cpp_check "" "" "defined(__SSE__)" && define __SSE__
     PROF_GEN_CC="${QPRE}prof-gen ${QPRE}prof-dir."
     PROF_GEN_LD=
     PROF_USE_CC="${QPRE}prof-use ${QPRE}prof-dir."
@@ -1106,8 +1106,8 @@
 
 filters="crop select_every"
 gpl_filters=""
-[ $swscale = yes ] && filters="resize $filters"
-[ $gpl = yes ] && filters="$filters $gpl_filters"
+[ $swscale = "yes" ] && filters="resize $filters"
+[ $gpl = "yes" ] && filters="$filters $gpl_filters"
 
 cat > conftest.log <<EOF
 platform:      $ARCH
