--- x2641820.org/configure	2010-12-07 20:37:30.000000000 +0900
+++ x2641820/configure	2010-12-07 20:40:42.000000000 +0900
@@ -387,7 +387,7 @@
             define HAVE_ALTIVEC
             AS="${AS-${cross_prefix}gcc}"
             if [ $SYS = MACOSX ] ; then
-                CFLAGS="$CFLAGS -faltivec -fastf -mcpu=G4"
+                CFLAGS="$CFLAGS -faltivec -fastf -mcpu=G4 -mtune=7450"
             else
                 CFLAGS="$CFLAGS -maltivec -mabi=altivec"
                 define HAVE_ALTIVEC_H
@@ -818,8 +818,8 @@
 
 filters="crop select_every"
 gpl_filters=""
-[ $swscale = yes ] && filters="resize $filters"
-[ $gpl = yes ] && filters="$filters $gpl_filters"
+[ "$swscale" = "yes" ] && filters="resize $filters"
+[ "$gpl" = "yes" ] && filters="$filters $gpl_filters"
 
 cat > conftest.log <<EOF
 Platform:   $ARCH
