--- x2641913.org/configure	2011-02-19 22:10:08.000000000 +0900
+++ x2641913/configure	2011-02-19 22:23:12.000000000 +0900
@@ -392,7 +392,7 @@
             define HAVE_ALTIVEC
             AS="${AS-${cross_prefix}gcc}"
             if [ $SYS = MACOSX ] ; then
-                CFLAGS="$CFLAGS -faltivec -fastf -mcpu=G4"
+                CFLAGS="$CFLAGS -faltivec -fastf -mcpu=G4 -mtune=7450"
             else
                 CFLAGS="$CFLAGS -maltivec -mabi=altivec"
                 define HAVE_ALTIVEC_H
@@ -843,8 +843,8 @@
 
 filters="crop select_every"
 gpl_filters=""
-[ $swscale = yes ] && filters="resize $filters"
-[ $gpl = yes ] && filters="$filters $gpl_filters"
+[ "$swscale" = "yes" ] && filters="resize $filters"
+[ "$gpl" = "yes" ] && filters="$filters $gpl_filters"
 
 cat > conftest.log <<EOF
 Platform:   $ARCH
