--- x2641659.org/configure	2010-06-25 18:09:03.000000000 +0900
+++ x2641659/configure	2010-06-25 18:17:48.000000000 +0900
@@ -333,7 +333,7 @@
       define HAVE_ALTIVEC
       AS="${AS-${cross_prefix}gcc}"
       if [ $SYS = MACOSX ] ; then
-        CFLAGS="$CFLAGS -faltivec -fastf -mcpu=G4"
+        CFLAGS="$CFLAGS -faltivec -fastf -mcpu=G4 -mtune=7450"
       else
         CFLAGS="$CFLAGS -maltivec -mabi=altivec"
         define HAVE_ALTIVEC_H
