--- x2641643.org/configure	2010-06-11 13:24:47.000000000 +0900
+++ x2641643/configure	2010-06-11 13:35:01.000000000 +0900
@@ -333,7 +333,7 @@
       define HAVE_ALTIVEC
       AS="${AS-${cross_prefix}gcc}"
       if [ $SYS = MACOSX ] ; then
-        CFLAGS="$CFLAGS -faltivec -fastf -mcpu=G4"
+        CFLAGS="$CFLAGS -faltivec -fastf -mcpu=G4 -mtune=7450"
       else
         CFLAGS="$CFLAGS -maltivec -mabi=altivec"
         define HAVE_ALTIVEC_H
