--- x2642057.org/configure	2011-08-10 23:57:27.000000000 +0900
+++ x2642057/configure	2011-08-11 08:14:38.000000000 +0900
@@ -526,7 +526,7 @@
             define HAVE_ALTIVEC
             AS="${AS-${cross_prefix}gcc}"
             if [ $SYS = MACOSX ] ; then
-                CFLAGS="$CFLAGS -faltivec -fastf -mcpu=G4"
+                CFLAGS="$CFLAGS -faltivec -fastf -mcpu=G4 -mtune=7450"
             else
                 CFLAGS="$CFLAGS -maltivec -mabi=altivec"
                 define HAVE_ALTIVEC_H
@@ -1076,8 +1076,8 @@
 
 filters="crop select_every"
 gpl_filters=""
-[ $swscale = yes ] && filters="resize $filters"
-[ $gpl = yes ] && filters="$filters $gpl_filters"
+[ "$swscale" = "yes" ] && filters="resize $filters"
+[ "$gpl" = "yes" ] && filters="$filters $gpl_filters"
 
 cat > conftest.log <<EOF
 Platform:   $ARCH
